### PR TITLE
Feature: AlertViewData - Adds option for a secondary alert button

### PR DESCRIPTION
**Pull request by PetterSW**
# Why?

In cases where we want to have a second option that is not "cancel"

# What?

Adds optional parameters specifying an alert button 

# Version Change

Minor change

# UI Changes

Skjermbilde 2022-08-17 kl  12 56 22

### CODE REVIEW

File: Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift [hunk 1]
```diff
@@ -43,21 +52,43 @@ public struct AlertViewData: Identifiable, Equatable {
...
+        )
 
-        case .destructive:
-            primaryButton = .destructive(Text(viewData.actionTitle), action: viewData.action)
-        }
+        let secondaryButton = Self.createActionButton(
+            style: viewData.secondaryButtonStyle,
+            title: viewData.secondaryActionTitle,
+            action: viewData.secondaryAction
+        )
```
Comment (fespinoza) [inline Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift:65]: I think it's better to do something like
```suggestion
        let secondaryButton: Alert.Button
        if let secondaryActionTitle = viewData.secondaryActionTitle, let secondaryAction = viewData.secondaryAction {
            secondaryButton = Self.createActionButton(
                style: viewData.secondaryButtonStyle,
                title: secondaryActionTitle,
                action: secondaryAction
            )
        } else {
            secondaryButton = .cancel()
        }
```

the `createActionButton` as is right now assumes that if any property is missing, you are building a `cancel` button

that's too much of an assumption IMO.

then it's better to localize that behavior to the secondary button alone instead.

File: Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift [hunk 2]
```diff
@@ -43,21 +52,43 @@ public struct AlertViewData: Identifiable, Equatable {
...
+        else { return .cancel() }
+
+        switch style {
+        case .default:
+            return .default(Text(title), action: action)
+
+        case .destructive:
+            return .destructive(Text(title), action: action)
+        }
+    }
```
Comment (fespinoza) [inline Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift:93]:
```suggestion
    private static func createActionButton(
        style: AlertViewData.ActionButtonStyle,
        title: String,
        action: @escaping (() -> Void)
    ) -> Alert.Button {
        switch style {
        case .default:
            return .default(Text(title), action: action)

        case .destructive:
            return .destructive(Text(title), action: action)
        }
    }
```

**Review by fespinoza — APPROVED** [review]
Small changes proposed. Make sure the formatting is correct after apply the changes

File: Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift (hunk 1 above)
Comment (PetterSW) [reply Sources/SATSCore/Extensions/SwiftUI/ViewData/AlertViewData.swift]: Agree!

**Pull request merged**